### PR TITLE
fix: turn off `Xdebug` note when running phpstan

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
See below image
image

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by datamweb — COMMENTED** [review]
So that's why I got this NOTE
```console
> phpstan analyze src/Config/Routes.php

 ! [NOTE] The Xdebug PHP extension is active, but "--xdebug" is not used. This may slow down performance and the process
 !        will not halt at breakpoints.
```

**Review by paulbalandan — CHANGES_REQUESTED** [review]
Please add it also on test-phpstan.yml

**Review by datamweb — APPROVED** [review]

**Review by paulbalandan — APPROVED** [review]

**Comment by kenjis** [comment]
```console
$ vendor/bin/phpstan --help | grep xdebug
      --xdebug                                 Allow running with XDebug for debugging purposes
```

**Review by kenjis — CHANGES_REQUESTED** [review]
See https://github.com/codeigniter4/CodeIgniter4/pull/6851#issuecomment-1312840552

**Comment by ddevsr** [comment]
In locally, when i run `vendor/bin/phpstan analyze --xdebug` better faster without `--xdebug`. If in GH Action affect report result to Coverage, i'll remove changes `test-phpstan.yml`

**Comment by paulbalandan** [comment]
Reading into phpstan-src's code, phpstan actually turns off xdebug if active for better performance.

**Comment by ddevsr** [comment]
My env based on [README Tests](https://github.com/codeigniter4/CodeIgniter4/blob/develop/tests/README.md#requirements) to use `xdebug.mode` to `coverage`.

**Comment by paulbalandan** [comment]
> My env based on [README Tests](https://github.com/codeigniter4/CodeIgniter4/blob/develop/tests/README.md#requirements) to use `xdebug.mode` to `coverage`.

yes, that's for testing. in this case, `xdebug.mode` will be turned off only when running phpstan

File: admin/pre-commit [hunk 1]
```diff
@@ -25,9 +25,9 @@ if [ "$FILES" != "" ]; then
 
     # Run on whole codebase
     if [ -d /proc/cygdrive ]; then
-        ./vendor/bin/phpstan analyse
+        ./vendor/bin/phpstan analyse --xdebug
```
Comment (paulbalandan) [inline admin/pre-commit:28]:
```suggestion
        XDEBUG_MODE=off ./vendor/bin/phpstan analyse
```

File: admin/pre-commit [hunk 2]
```diff
@@ -25,9 +25,9 @@ if [ "$FILES" != "" ]; then
...
     else
-        php ./vendor/bin/phpstan analyse
+        php ./vendor/bin/phpstan analyse --xdebug
```
Comment (paulbalandan) [inline admin/pre-commit:30]:
```suggestion
        XDEBUG_MODE=off php ./vendor/bin/phpstan analyse
```

File: composer.json [hunk 3]
```diff
@@ -77,7 +77,7 @@
             "bash -c \"if [ -f admin/setup.sh ]; then bash admin/setup.sh; fi\""
         ],
         "analyze": [
-            "phpstan analyze",
+            "phpstan analyze --xdebug",
```
Comment (paulbalandan) [inline composer.json:80]:
```suggestion
            "bash -c \"XDEBUG_MODE=off phpstan analyse\"",
```

File: .github/workflows/test-phpstan.yml [hunk 4]
```diff
@@ -77,4 +77,4 @@ jobs:
         run: composer update --ansi --no-interaction
 
       - name: Run static analysis
-        run: vendor/bin/phpstan analyse
+        run: vendor/bin/phpstan analyse --xdebug
```
Comment (paulbalandan) [inline .github/workflows/test-phpstan.yml:80]: Please reset this change

**Review by paulbalandan — CHANGES_REQUESTED** [review]

File: .github/workflows/test-phpstan.yml (hunk 4 above)
Comment (ddevsr) [reply .github/workflows/test-phpstan.yml:80]: No want disable Xdebug with `XDEBUG_MODE=off`?

Comment (paulbalandan) [reply .github/workflows/test-phpstan.yml:80]: Ah yes. In the Setup PHP step, add `coverage: none`

Comment (ddevsr) [reply .github/workflows/test-phpstan.yml]: Done

File: composer.json (hunk 3 above)
Comment (ddevsr) [reply composer.json]: Done

File: admin/pre-commit (hunk 2 above)
Comment (ddevsr) [reply admin/pre-commit]: Done

File: admin/pre-commit (hunk 1 above)
Comment (ddevsr) [reply admin/pre-commit]: Done

**Review by kenjis — APPROVED** [review]

**Review by paulbalandan — APPROVED** [review]

**Review by MGatner — APPROVED** [review]

**Pull request merged**